### ISSUE TITLE
AWS authentication with web identity token

### ISSUE THREAD

**Issue by hamza-tumturk** [laserdisc-io/fs2-aws#724]
Hi,
using this library for an app running on EKS with injecting AWS credentials via IRSA, it doesn't seem to read the credentials provided as Web identity Token. Were using the lib `fs2.aws.kinesis.publisher`.
These are provided via the env vars `AWS_ROLE_ARN` and `AWS_WEB_IDENTITY_TOKEN_FILE`.
Can we have support for this authentication method as well, please?
Thanks

**Comment by barryoneill** [comment]
This may or not be related to your problem, but I know when I tried to use fs2-aws with an OIDC token I had similar problems. 

Despite the AWS SDK itself in use being v2,  the `amazon-kinesis-producer` was still dependent on the AWS SDK v1 (see https://github.com/awslabs/amazon-kinesis-producer/blob/master/java/amazon-kinesis-producer/pom.xml#L60) 

To get OIDC to work, I needed to explicitly pull the V1 STS provider as a dependency, e.g.

```
 "com.amazonaws" % "aws-java-sdk-sts" % "1.12.11" % "runtime"
```

Maybe that's what's happening for you?
